project_classification/__init__.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Author: Joël Grand-guillaume (Camptocamp)
+#    Copyright 2011 Camptocamp SA
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from . import project_classification
+import psycopg2
+
+
+def fill_classification_id(cr):
+        query_nb_records = 'SELECT count(id) from project_project;'
+        cr.execute(query_nb_records)
+        res = cr.fetchall()
+        if res[0][0] != 0:
+            try:
+                query_classification_id = (
+                    'SELECT count(id) from project_project '
+                    'where classification_id is null;'
+                    )
+                cr.execute(query_classification_id)
+                res2 = cr.fetchall()
+                if res[0][0] == res2[0][0]:
+                    # all project have a classification_id set
+                    return
```

Comment:
I brought the confusion, but I think it is safe to replace the conditionshere by this one:

```python
cr.execute("select exists(select * from information_schema.tables where table_name='project_classification')")
if not cr.fetchone():
   # all the stuff in the `except` block
```
Because if the table does not exist, the classification_id field neither (and further, as we install this module, the table should never happen to exist prior to the installation but we can keep the condition as a safeguard).